src/utils/sample.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * Returns a randomly-selected item from an array
+ * @template T
+ * @param {T[]} arr Array of values to be sampled
+ * @param {Number} [randomValue] random value that can be injected for more deterministic behavior
+ * @return {T=} an item from the array (or undefined if empty array)
+ */
+function sample(arr, randomValue = Math.random()) {
```

Comment:
I had to double-check when default arguments are evaluated. I think in Python they're evaluated at definition time, in which case every invocation of `sample` would get the same value. But you're correct, in Javascript they're evaluated at call time. 🎉